src/main/java/com/weather/assignment/initializer/WeatherDataInitializer.java

```diff
@@ -0,0 +1,18 @@
+package com.weather.assignment.initializer;
+
+import com.weather.assignment.service.WeatherService;
+import org.springframework.boot.CommandLineRunner;
+
+public class WeatherDataInitializer implements CommandLineRunner {
+
+  private final WeatherService weatherService;
+
+  public WeatherDataInitializer(WeatherService weatherService) {
+    this.weatherService = weatherService;
+  }
+
+  @Override
+  public void run(String... args) throws Exception {
+    weatherService.getWeatherDetails();
```

Comment:
As discussed in the call, this method name is not true to its purpose. Please change it.